sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4993,11 +5116,16 @@ pi_result piEnqueueMemBufferMap(pi_queue Queue, pi_mem Buffer,
   if (Buffer->MapHostPtr) {
     *RetMap = Buffer->MapHostPtr + Offset;
   } else {
-    ze_host_mem_alloc_desc_t ZeDesc = {};
-    ZeDesc.flags = 0;
+    if (IndirectAccessTrackingEnabled) {
+      // Use the version with reference counting
+      PI_CALL(piextUSMHostAlloc(RetMap, Queue->Context, nullptr, Size, 1));
```

Comment:
Similar here: can't we always take this path?

Reply:
We can take this path always, I've put in under guard because all changes of this PR are under the guard.